### DISCUSSION TITLE
Cannot complete browser.url(*) due to an alert (modal)

### DISCUSSION THREAD

**Discussion by yzvezdin** [webdriverio/webdriverio#7332] [Technical Discussion]
Hi colleagues,
I have an issue while calling broser.url(*). We have an alert to confirm that you want to leave the page without saving data. And it seems to be _modal window_. So the callback is waiting for some event to happen and it is not. And you cannot overdue it.
I was thinking on some workaround on making method that will reduce Page timeout to 100ms then navigating and checking for a modal window. But not sure it is a good workaround for every navigation :(
Maybe you can embed some verifications into the native _url_ method?

**Comment by christian-bromann** [comment]
> Maybe you can embed some verifications into the native _url_ method?

You mean, check if an alert is open when calling the url command. I believe this would already fail with a decent error message, am I right?

**Reply by yzvezdin** [reply]
In our case it fails by timeout :( so no descent message or workaround. only making some helper method that will process such scenario instead of browser.url()

**Reply by christian-bromann** [reply]
which browser and automation protocol?

**Reply by yzvezdin** [reply]
It is Chrome with Jasmine + WDIO using TypeScript.

**Reply by christian-bromann** [reply]
Are you using Chromedriver?

**Reply by yzvezdin** [reply]
We use "wdio-chromedriver-service": "^7.2.0"

**Reply by christian-bromann** [reply]
From what I see, the behavior is that the url command fails but triggers to close the confirm modal. The failed url command is then automatically retried which then causes a successful page transition.

**Reply by yzvezdin** [reply]
Unfortunately in our case it doesn't close any alerts/windows so it hangs for the timeout set and then test fails.
The issue is - the alert remains open and blocks other tests as well (we can make verification for after each test, of course but that seems like a hook again)
Thanks for you time!

**Reply by christian-bromann** [reply]
I am not sure why you can't reproduce this but this test works without error:
```ts
describe('My Login application', () => {
    it('should login with valid credentials', async () => {
        await browser.url('http://json.org')
        await browser.execute(() => setTimeout(() => confirm("JI"), 3000))
        await browser.pause(4000)
        await browser.url('https://webdriver.io')
        console.log(await browser.getTitle());
    });
});
```

For both automation protocols.

**Reply by yzvezdin** [reply]
We get a message like this so navigation never happens and... browser.url() never resolved... .and we don't get
![image](https://user-images.githubusercontent.com/16023406/130797671-58773880-9fa8-45dc-83e5-c5df9d8ee775.png)
Following error:
'[4:20 PM] Iurii Zviezdin
    Error: Timeout - Async function did not complete within 100000ms (set by jasmine.DEFAULT_TIMEOUT_INTERVAL)
    at <Jasmine>
    at listOnTimeout (internal/timers.js:549:17)
    at processTimers (internal/timers.js:492:7)'
![image](https://user-images.githubusercontent.com/16023406/130797959-7e922ecd-0b59-46de-9cf8-6f47d34fad56.png)

**Comment by christian-bromann** [comment]
@yzvezdin this seems to be a bug when using the devtools protocol. You can workaround by starting Chromedriver or using `wdio-chromedriver-service`. See #7337

**Reply by yzvezdin** [reply]
Thank you! 
Have a great day!
